src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -240,179 +208,209 @@ def _merge(self, other: "PartialSnapshot") -> "PartialSnapshot":
             self._forward_model_states[forward_model_id].update(other_fm_data)
         return self
 
-    def from_cloudevent(self, event: CloudEvent) -> "PartialSnapshot":
+    def update_from_cloudevent_and_generate_update_snapshot(
+        self,
+        event: CloudEvent,
+        update_event_snapshot: Optional["Snapshot"] = None,
+    ) -> "Snapshot":
+        """Add the data from the cloudevent, and also generate a new snapshot containing only the updated data.
+
+        Args:
+            event (CloudEvent): The CloudEvent with the data to update.
+            update_event_snapshot (Optional[&quot;Snapshot&quot;], optional): The snapshot which should only have the updated data attached. Returns a new snapshot if None is provided.
+
+        Returns:
+            Snapshot: A snapshot only containing the updated data.
```

Comment:
If provided, it will update the input snapshot. This is how it was done before, so we could aggregate all cloudevents into a single snapshot without having to explicitly merge it into a snapshot.
I do not like this, but changing it would change functionality and require alterations to a lot of tests. Should I do it in this PR?

Reply:
No.